docs/data/toolpad/core/components/app-provider/app-provider.md

```diff
@@ -6,20 +6,90 @@ components: AppProvider
+<p class="description">The app provider component provides the necessary context to easily set up a Toolpad application.</p>
+
+By wrapping an application at the root level with an `AppProvider` component, many of Toolpad's features (such as routing, navigation and theming) can be automatically enabled to their fullest extent, abstracting away complexity and helping you focus on the details that matter.
+
+It is not mandatory that every application is wrapped in an `AppProvider`, but it is highly recommended for most apps that use Toolpad.
 
 ## Basic
 
-Wrap the whole application with the `AppProvider` component to enable many of Toolpad's features.
+Wrap an application page with the `AppProvider` component.
+
+Ideally, the `AppProvider` should wrap every page in the application, therefore in most projects it should be imported and placed in the file that defines a **shared layout** for all pages.
+
+In the following (purely demonstrative) example, an `AppProvider` component wrapping the page provides it with a default theme, and a `DashboardLayout` placed inside it gets its navigation and routing features automatically set based on the props passed to the `AppProvider`.
 
 {{"demo": "AppProviderBasic.js", "height": 500, "iframe": true}}
 
 ## Next.js
 
-The `AppProvider` for Next.js applications includes routing out-of-the-box.
+The `AppProvider` for Next.js applications includes some Next.js integrations out-of-the-box.
+
+By using the specific `AppProvider` for Next.js you do not have to manually configure the integration between some Toolpad features and the corresponding Next.js features (such as routing), making the integration automatic and seamless.
 
 ```tsx
 import { AppProvider } from '@toolpad/core/nextjs/AppProvider';
 // or
 import { AppProvider } from '@toolpad/core/nextjs';
 ```
+
+### Next.js App Router
+
+When using the **Next.js App Router**, the most typical file where to import and use `AppProvider` will be at the top level `layout.tsx` file that defines the layout for all the application pages.
+
+```tsx
+// app/layout.tsx
+
+import { AppProvider } from '@toolpad/core/nextjs/AppProvider';
+
+export default function Layout(props) {
+  const { children } = props;
+
+  return (
+    <html lang="en">
+      <body>
+        <AppProvider>{children}</AppProvider>
+      </body>
+    </html>
+  );
+}
+```
+
+### Next.js Pages Router
+
+When using the **Next.js Pages Router**, the most typical file where to import and use `AppProvider` in order to wrap every page in the application will be the `pages/_app.tsx` file.
+
+```tsx
+// pages/_app.tsx
+
+import { AppProvider } from '@toolpad/core/nextjs/AppProvider';
+
+export default function App(props) {
+  const { Component, pageProps } = props;
+
+  return (
+    <AppProvider>
+      <Component {...pageProps} />
+    </AppProvider>
+  );
+}
+```
+
+## Theming
```

Comment:
I would also add a separate "Navigation" section that explains the `navigation` prop and an example of a hierarchy that is a bit more extensive with subheaders and nested items. Ideally something realistic for a dashboard that end users could relate to.

This would also be fine to do in follow-up